draft-rescorla-tls-esni.md

```diff
@@ -188,24 +188,18 @@ structure, defined below.
 checksum
 : First four (4) octets of the SHA-256 message digest {{RFC6234}} of the
 ESNIKeys structure starting from the first octet of "keys" to the end of
-the stucture.
+the structure.
 
 keys
 : The list of keys which can be used by the client to encrypt the SNI.
 Every key being listed MUST belong to a different group.
 
 padded_length
-: The length to pad the ServerNameList value to prior to encryption.
-This value SHOULD be set to the largest ServerNameList the server
-expects to support rounded up the nearest multiple of 16.
-{:br}
-
-[[OPEN ISSUE: An alternative to padding is to instead send
-a hash of the server name. This would be fixed-length, but
-have the disadvantage that the server has to retain a table
-of all the server names it supports, and will not work if
-the mapping between the client-facing server and hidden 
-server uses wildcards.]]
+: 
+The length to pad the ServerNameList value to prior to encryption. 
+This value SHOULD be set to the largest ServerNameList the server 
+expects to support rounded up the nearest multiple of 16. If the 
+server supports wildcard names, it SHOULD set this value to 256.
```

Comment:
Considering the fact that the distribution of the ESNI keys is not limited to DNS, it might be worth specifying that the 256 chars limitation is specific to when DNS is being used; e.g. "If the server supports wildcard names *resolved by DNS*".

Reply:
👍